### DISCUSSION TITLE
登録セッションのカレンダー同期機能

### DISCUSSION THREAD

**Discussion by jacopen** [cloudnativedaysjp/dreamkast#1355] [Ideas]

**Comment by jacopen** [comment]
このあたりのGem活用していけないかな

https://github.com/icalendar/icalendar

**Comment by jacopen** [comment]
メモ: Twitterで要望あり

**Comment by jacopen** [comment]
あるといいカレンダー
- イベント全体のスケジュールをical形式で提供
  - イベントのTalksでshow_on_timetableが1なものを全部引っ張ってきてicalendar gemにぶっ込むだけでいけそう？
- 参加者が参加登録したものをpersonalizeされたical形式で提供
  - https://event.cloudnativedays.jp/cnsec2022/profiles/calendar.ics?code=random value  みたいなURLでアクセスさせる
  - random valueで簡易な認証させる。random valueはユーザーが自身のプロファイルから再作成可能にする

**Comment by ShotaKitazawa** [comment]
ちなみにこの話は、 RDB に接続しに行ければ dreamkast である必要はない (別プロセス/別リポジトリで管理されたソースコードで良い) という理解で合っていますか？

**Reply by jacopen** [reply]
Yesです！　ただ別プロセスからRDBを共有して切り出すというアーキテクチャが適しているかどうかは議論の余地ありかもですが
